scripts/installer/ec2-create-instance.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+# For docs, see the "Deployment" page in the Dev Guide.
+
+SUGGESTED_REPO_URL='https://github.com/IQSS/dataverse.git'
+SUGGESTED_BRANCH='develop'
+
+usage() {
+  echo "Usage: $0 -r $REPO_URL -b $SUGGESTED_BRANCH" 1>&2
+  exit 1
+}
+
+REPO_URL=$SUGGESTED_REPO_URL
+
+while getopts ":r:b:" o; do
+  case "${o}" in
+  r)
+    REPO_URL=${OPTARG}
+    ;;
+  b)
+    BRANCH_NAME=${OPTARG}
+    ;;
+  *)
+    usage
+    ;;
+  esac
+done
+
+AWS_CLI_VERSION=$(aws --version)
+if [[ "$?" -ne 0 ]]; then
+  echo 'The "aws" program could not be executed. Is it in your $PATH?'
+  exit 1
+fi
+
+if [ "$BRANCH_NAME" = "" ]; then
+  echo "No branch name provided. You could try adding \"-b $SUGGESTED_BRANCH\" or other branches listed at $SUGGESTED_REPO_URL"
+  usage
+  exit 1
+fi
+
+if [[ $(git ls-remote --heads $REPO_URL $BRANCH_NAME | wc -l) -eq 0 ]]; then
+  echo "Branch \"$BRANCH_NAME\" does not exist at $REPO_URL"
+  usage
+  exit 1
+fi
+
+SECURITY_GROUP='dataverse-sg'
+GROUP_CHECK=$(aws ec2 describe-security-groups --group-name $SECURITY_GROUP)
+if [[ "$?" -ne 0 ]]; then
+  echo "Creating security group \"$SECURITY_GROUP\"."
+  aws ec2 create-security-group --group-name $SECURITY_GROUP --description "security group for Dataverse"
+  aws ec2 authorize-security-group-ingress --group-name $SECURITY_GROUP --protocol tcp --port 22 --cidr 0.0.0.0/0
+  aws ec2 authorize-security-group-ingress --group-name $SECURITY_GROUP --protocol tcp --port 80 --cidr 0.0.0.0/0
+  aws ec2 authorize-security-group-ingress --group-name $SECURITY_GROUP --protocol tcp --port 443 --cidr 0.0.0.0/0
+  aws ec2 authorize-security-group-ingress --group-name $SECURITY_GROUP --protocol tcp --port 8080 --cidr 0.0.0.0/0
+fi
+
+RANDOM_STRING="$(uuidgen | cut -c-8)"
+KEY_NAME="key-$USER-$RANDOM_STRING"
```

Comment:
I wonder if adding the branch name to this would provide value as well. Not a blocker for this version though

Reply:
We could, I guess. I was trying to keep the keyname somewhat short. My thought was that we could always check the running instance to see what branch is, if necessary.